### PR TITLE
Unifying @function and @contractor

### PR DESCRIPTION
**Pull request by yashvardhan747**
Solves [#113](https://github.com/JuliaIntervals/IntervalConstraintProgramming.jl/issues/113)

### CODE REVIEW

File: src/functions.jl [hunk 1]
```diff
@@ -14,8 +14,32 @@ mutable struct ConstraintFunction{F <: Function, G <: Function}
     expression::Expr
 end
 
-function (C::ConstraintFunction)(args...)
-    return C.forward(args)[1]
+function (C::ConstraintFunction)(args...) #To access forward function call f(X) and for backward function call f(A,X)
+                                          #where A is constraint
+
+    if length(args)==1
```
Comment (dpsanders) [inline src/functions.jl]: You should do this via dispatch instead.

Comment (yashvardhan747) [reply src/functions.jl]: okay..!!

**Comment by dpsanders** [comment]
Please use

git rebase master

instead of merging.

**Comment by dpsanders** [comment]
How does this solve #113?

**Pull request closed without merging**